core/src/main/scala/org/apache/spark/deploy/master/Master.scala

```diff
@@ -702,8 +712,6 @@ private[deploy] class Master(
     worker.addExecutor(exec)
     worker.endpoint.send(LaunchExecutor(masterUrl,
       exec.application.id, exec.id, exec.application.desc, exec.cores, exec.memory))
-    exec.application.driver.send(ExecutorAdded(
-      exec.id, worker.id, worker.hostPort, exec.cores, exec.memory))
```

Comment:
I think it would be better to notify application of `ExecutorAdded` only when worker reports back executor state to `RUNNING`. Otherwise application will possibly receive `ExecutorAdded` while worker is failed to add executor. What do you think?